### PR TITLE
Added Topic About Dating Experience

### PR DESCRIPTION
**Pull request by Dank-Elf-Boy**
This topic covers the player's previous relationship experiences they've acquired in their lifetime and how it can help strengthen the relationship the player has with Monika (Issue: #2694). I haven't tried adding topics in a while so go easy on me haha. And please don't outright withdraw the request if a few things are off, just let me know and I can fix them. Thank you.

### CODE REVIEW

**Comment by jmwall24** [comment]
Hi, thanks for contributing!  Off the bat I notice you have 19 invalid sprites:
screenshot_101

You can check here [sprite-chart](https://github.com/Monika-After-Story/MonikaModDev/blob/master/Monika%20After%20Story/game/sprite-chart.rpy) and use ctrl-f to search thru and see what sprites are currently available.  If there are some that don't exist that you really have to have, you can request them and we'll look at them.  But I'm sure you can find expressions that fit your needs for most of your topic.

**Comment by Rai99** [comment]
I saw your original issue suggesting/requesting the 2 topics and I thought I'd let you know that I was also working on this very topic you requested.

**Comment by Rai99** [comment]
I guess I'll speed it up and pick up the pace.

**Comment by Bam8647** [comment]
Haha it's okay Rai, this one was pretty important to me so I thought I'd give it a shot at contributing at least once to this mod. (: 

The "Are You A Virgin" topic is a little iffy at the moment but if your up to the task then I'd say go for it. Just make it convincing enough to let it hopefully pass lol

**Comment by jmwall24** [comment]
Will leave an actual code review later, but you need to rebase this to `content` branch.  Look here to see how to do that https://github.com/Monika-After-Story/MonikaModDev/wiki/FAQ#i-put-my-pull-request-in-the-wrong-branch-and-dont-want-to-suffer-the-wrath-of-devs-how-do-i-fix-it

**Comment by Rai99** [comment]
I mean, I'm in the middle of writing my version of this and I still plan to put up a PR for it because it's almost done and what else am I supposed to do with it? 

I could take a look at your version here in a bit and help clean it up, but I do have to say that content wise, it's a **fair** bit better than the last contribution I looked at.

**Comment by Bam8647** [comment]
Wow I missed a lot of quotations **oof**.

I'm fixing the mistakes now, once I'm finished to I continue under branch patch-4? Or create a new pull request. I have two windows up on github and I switched my forked branch to merge to **content** instead of **master** all while having another window up of me fixing the changes in my patch

**Comment by multimokia** [comment]
Yep, just commit to patch 4, we'd like to keep it all in one PR.

**Comment by multimokia** [comment]
Looks like you've still got some invalid sprites there, Travis gives line numbers, but it's not always correct with them...

**Comment by Bam8647** [comment]
Yeah i think it's the beginning of the topic let me check again :P

**Comment by multimokia** [comment]
looks like line 7262, sprite 1ekb

**Comment by Bam8647** [comment]
Was that the only one? Thank god btw you're able to tell which sprite, I've been going back and forth with the expression viewing in MAS lmao..

**Comment by multimokia** [comment]
Haha, don't worry about it. Travis reported that on line 7339, and that's not even covered in your changes. Good job, Travis.

File: Monika After Story/game/script-topics.rpy [hunk 1]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+init 5 python:
+    addEvent(Event(persistent.event_database,"monika_dating",prompt="Dating Experience",category=['you','commitment'],random=True))
+    
+label monika_dating:
+    m 2eud "You know I've been really curious lately, [player]..."
+    m 2eka "We've been talking for so long that now I think would be the best time to ask you this important question..."
+    m 1eud "How much dating experience do you have?"
+    m 1lud "Like... have you ever been in a relationship before?"
+    menu:
+       m "Maybe more than once?"
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: This bit here should be lined up with the lines following it

Comment (Bam8647) [reply Monika After Story/game/script-topics.rpy]: You mean align **m "Maybe more than once?"** under menu: or have the lines after the question under it?

Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: `m "Maybe more than once?"` should be aligned with the menu options following it.

**Comment by multimokia** [comment]
You'll also want to move that return at the end back a full indent, and not a space

**Comment by Rai99** [comment]
It seems that we took a slightly different approach to the topic, but I think you did a good job writing up the topic and covering what you wanted to cover.

**Comment by multimokia** [comment]
Hey there we go! Passed Travis

**Comment by Bam8647** [comment]
Wahoo!!

File: Monika After Story/game/script-topics.rpy [hunk 2]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 3hksdlb "Sorry I lost track of what I was going to say next ahaha!"
+            m 1ekbfa "I could go on about how wonderful you are [player] ehehe. Just know this though."
+            m 2ekbfa "No matter how many heart breaks you've been through, I'll always be here for you."
+            m 5ekbfa "Our soul searching is finally over. I'll be yours forever [player]!"
+            m 5ekbfa "Will you be mine?"
+        
+        "Yes, but only once":
+            m 1eka "Ah, not a lot of experience huh?"
+            m 2eka "That's okay [player], I can relate too so don't worry"
+            m 3ekb "Yeah I may appear like a girl who gets all the guys but really I don't ahaha!"
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: I would add a comma after "Yeah", "guys" and possibly "don't".

File: Monika After Story/game/script-topics.rpy [hunk 3]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 2lksdlb "Especially with how occupied I've kept myself over the years I just never had the time"
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: This line doesn't flow very well, but adding a comma after "years" or breaking the last part onto a new line would likely fix it. 

Missing period.

File: Monika After Story/game/script-topics.rpy [hunk 4]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 2eka "But now that I do I think it's safe to say I'm ready to start something special..."
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: Add a comma after "do".

I would also add "that" after "say".

File: Monika After Story/game/script-topics.rpy [hunk 5]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 5ekbfa "With you [player]..."
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: Add a comma after "you".

File: Monika After Story/game/script-topics.rpy [hunk 6]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 5ekbfa "Are you ready?"
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: Body pose 5 usually follows a dissolve transition as mentioned above.

File: Monika After Story/game/script-topics.rpy [hunk 7]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+        
+        "No, you're my first":  
+            m 6wubsw "What?! I-I'm your first?!"
+            m 1tubfb "Oh.. I see."
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: Ellipses are usually 3 periods long.

You could also add a timed pause here.

File: Monika After Story/game/script-topics.rpy [hunk 8]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 1tfb "You're just saying that to make me feel extra special huh [player]?"
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: I would replace "huh" with "aren't you,"

Don't forget the comma before [player]

File: Monika After Story/game/script-topics.rpy [hunk 9]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 1tku "There's no way someone like you has never dated before, I mean look at you!"
+            m 2hubfb "You're the definition of cute and sweetness!"
+            m 2ekbfa "Well.. If you're not just messing with me and actually telling me the truth then.."
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: I think ellipses are usually 3 periods.

File: Monika After Story/game/script-topics.rpy [hunk 10]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 1ekbfb "I'm honored to be your first [player]!"
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: Add a comma after "first".

File: Monika After Story/game/script-topics.rpy [hunk 11]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 5ekbfa "I hope to be your one and only."
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: Missing dissolve transition.

File: Monika After Story/game/script-topics.rpy [hunk 12]
```diff
@@ -7235,6 +7235,50 @@ label monika_hydration:
...
+            m 2lksdlb "Especially with how occupied I've kept myself over the years I just never had the time"
+            m 2eka "But now that I do I think it's safe to say I'm ready to start something special..."
+            m 5ekbfa "With you [player]..."
+            m 5ekbfa "Are you ready?"
+        
+        "No, you're my first":  
+            m 6wubsw "What?! I-I'm your first?!"
+            m 1tubfb "Oh.. I see."
+            m 1tfb "You're just saying that to make me feel extra special huh [player]?"
+            m 1tku "There's no way someone like you has never dated before, I mean look at you!"
```
Comment (Rai99) [inline Monika After Story/game/script-topics.rpy]: I think this comma could be replaced with a hyphen.

**Comment by Rai99** [comment]
Congratulations for passing Travis. I still need to wait for some expressions to be added for mine to pass.

File: Monika After Story/game/script-topics.rpy (hunk 11 above)
Comment (Bam8647) [reply Monika After Story/game/script-topics.rpy]: How do you do add that?

Comment (Rai99) [reply Monika After Story/game/script-topics.rpy]: show monika (The expression you want) at t11 zorder MAS_MONIKA_Z with dissolve
m (The expression you want) "(Dialogue)"

Don't forget to do the same when transitioning out of body pose 5.

Comment (Bam8647) [reply Monika After Story/game/script-topics.rpy]: I've never gotten that far figuring out how to do that haha.. I may need an exact format and what lines to add these on as I have no idea how this is done. I get my help by looking at similar topic formats to help me that's all I know lol..

**Comment by ThePotatoGuy** [comment]
okay need to figure out if this and #2721 can both be in or only one in.

**Comment by Bam8647** [comment]
@ThePotatoGuy
Well my version is to basically capture the player's experience with _dating in general_ while Rai's version delves more into relationship history. Both of our topics intersect when we list the responses except Rai added additional  "A few" selection.

My main focus of this topic is to capture **3 basic levels of experience: Well experienced, Experienced, and No experience** in regards to dating.

We have two options to settle this.

**1.** _We consolidate the topics and simply choose to have either my short and simple version or Rai's more extended version._

**2.** _One of us revamps our topics to help differentiate the two topics and possibly make our topics unlocked at different affection levels. Rai could remove the choices he has initially and make his topic about "The Experience of Past Relationships" while mine is to help Monika get a first impression on our "Dating History"._

**Comment by Rai99** [comment]
Looking back, I actually think I might need another option for if the player doesn't want to talk about it. I think that's an important option to have available, but 5 options is really pushing it. 

> My main focus of this topic is to capture 3 basic levels of experience: Well experienced, Experienced, and No experience in regards to dating.

That clears up a bit for me with your topic. One thing I wasn't sure of was that there was kind of a missing space with options. What if the player has been in 2 relationships or something? That's more than one, but I  wouldn't call that plenty. Also, there's the possible scenario where the player doesn't want to talk about it. 

With the way you have your options worded currently, it looks more about the number of relationships rather than experience. 

Personally, I don't think both of our topics can make it in.

**Comment by Bam8647** [comment]
I'll have to disagree. I believe our topics can still be used for an early inquiry of the player's "familiarity with dating" and then what we learned through our "trial and errors" of past relationships. We may need to change up some things as to not sound similar but I still think we can work this out. My topic would be prompted first where the player tells Monika their experience with asking other girls out followed by your topic where she would then question why our past relationships didn't work out. 

Again if the final say is that our topics sound redundant then we could always combine our topics.

**Pull request merged**